relayer/crates/starknet-chain-components/src/impls/messages/create_client.rs

```diff
@@ -73,10 +74,23 @@ where
             chain_id: payload.client_state.chain_id,
         };
 
+        // Convert Vec<u8> to a slice of 8 u32 values.
+        let mut root_u32: Vec<u32> = Vec::new();
+        for chunk_u8 in root.chunks_exact(4) {
+            let value_u32 = (chunk_u8[0] as u32)
+                | ((chunk_u8[1] as u32) << 8)
+                | ((chunk_u8[2] as u32) << 16)
+                | ((chunk_u8[3] as u32) << 24);
```

Comment:
probably we can use `from_le_bytes` here. but it takes `[u8; 4]`.

Reply:
ah you fixed it in 7d99039.

Reply:
Done in 7d99039a36e90d1827be4b9a033de9a3172a6b1c
But if the root is not of 32 elements, there will be an issue when comparing it with the [u32; 8] hash in Cairo, no?